lib/IRGen/GenDecl.cpp

```diff
@@ -3649,34 +3651,62 @@ IRGenModule::emitDirectRelativeReference(llvm::Constant *target,
   return relativeAddr;
 }
 
-/// Emit the protocol descriptors list and return it.
-llvm::Constant *IRGenModule::emitSwiftProtocols() {
-  if (SwiftProtocols.empty())
-    return nullptr;
-
-  // Define the global variable for the protocol list.
-  ConstantInitBuilder builder(*this);
-  auto recordsArray = builder.beginArray(ProtocolRecordTy);
-
-  for (auto *protocol : SwiftProtocols) {
-    auto record = recordsArray.beginStruct(ProtocolRecordTy);
-
-    // Relative reference to the protocol descriptor.
-    auto descriptorRef = getAddrOfLLVMVariableOrGOTEquivalent(
-                                   LinkEntity::forProtocolDescriptor(protocol));
-    record.addRelativeAddress(descriptorRef);
+static void appendLLVMUsedConditionalEntry(IRGenModule &IGM,
+                                           llvm::GlobalVariable *var,
+                                           llvm::Constant *ref) {
+  llvm::Metadata *metadata[] = {
+      // (1) which variable is being conditionalized, "target"
+      llvm::ConstantAsMetadata::get(var),
+      // (2) type, not relevant for a single-edge condition
+      llvm::ConstantAsMetadata::get(llvm::ConstantInt::get(
+          llvm::Type::getInt32Ty(IGM.Module.getContext()), 0)),
+      // (3) the "edge" that holds the target alive, if it's missing the target
+      // is allowed to be removed
+      llvm::ConstantAsMetadata::get(ref),
+  };
+  auto *usedConditional =
+      IGM.Module.getOrInsertNamedMetadata("llvm.used.conditional");
+  usedConditional->addOperand(
+      llvm::MDNode::get(IGM.Module.getContext(), metadata));
+}
 
-    record.finishAndAddTo(recordsArray);
-  }
+static void
+appendLLVMUsedConditionalEntry(IRGenModule &IGM, llvm::GlobalVariable *var,
+                               const ConformanceDescription &record) {
```

Comment:
Can you add a comment saying that `var` depends on record.protocol and record.type